### ISSUE TITLE
What does "Any IDs not assiend for figure node" mean?

### ISSUE THREAD

**Issue by jdemeyer** [sphinx-doc/sphinx#2483]
Apart from the obvious typo (`assiend`), it is really not clear what the message means. I'm guessing that it means that a figure does not have a number, but that's normal and there should not be a warning for this.

**Comment by tk0miya** [comment]
Thank you for reporting. Could you tell me how do you get the message? I think this message should be never shown.

**Comment by jdemeyer** [comment]
> Could you tell me how do you get the message?

It's quite complicated. It appears when compiling the SageMath documentation with Sphinx 1.4.1. But SageMath uses a very customized Sphinx, so it might be because of that.

**Comment by jdemeyer** [comment]
If it helps, the full message is

```
/usr/local/src/sage-config/src/doc/en/thematic_tutorials/sandpile.rst:None: WARNING: Any IDs not assiend for figure node
```

and [the document in question](https://github.com/sagemath/sage/blob/master/src/doc/en/thematic_tutorials/sandpile.rst) is one with quite a lot of `..figure ::` directives.

**Comment by tk0miya** [comment]
Indeed, it's very complicated the SageMath project. I tried to build the docs, but I cannot prepare the environment. I want to investigate where this warnings came from.
Could you give me a minimal sphinx project to reproduce the warnings?
It would be very helpful if we can build docs without building sage.

**Comment by lehmannro** [comment]
FWIW, the error comes from: [sphinx.writers.html.HTMLTranslator.add_fignumber](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/writers/html.py#L281).

**Comment by gplanansky** [comment]
Using the mac port sphinx-build 1.4 (py27) and mathjax, I get that "WARNING: Any IDs not assiend for figure node" message if, as below, I have a figure caption that leads with ":math:`x`".   

```
.. figure::  myfig.svg

   :math:`x`
```

If I prefix the figure caption with text, "foo:math:`x` as below, I get no such warning:

```
.. figure::  myfig.svg

   foo:math:`x`
```

If the latex expression is an empty field, ":math:``", as below, I also get no warning:

```
.. figure::  myfig.svg

   :math:``
```

**Comment by tk0miya** [comment]
@gplanansky Thank you for good information! I found the cause of warnings.
It comes from the figure having non text objects (like math) only.

I fixed it at 7ebe570.

Thank you all for reporting and helping!

**Comment by gplanansky** [comment]
You are most welcome, I am glad it helped track it down.  And, thank _you_,
very much, for so ably and so greatly contributing.
